src/mongocxx/private/collection.hpp

```diff
@@ -39,6 +39,13 @@ class collection::impl {
           client_impl(client) {
     }
 
+    impl(const impl& i) : collection_t{libmongoc::collection_copy(i.collection_t)} { }
```

Comment:
I would look at the source in the c driver. Are there any situations in which it returns a null pointer, for example?